File: hw/Vaca/u3/HW28SingletonAndAbstractFactory/HW28AbstractFactoryJavaScript/model/WinFactory.js
import GUIFactory from './GUIFactory.js';
import WinButton from './WinButton.js';
import WinMenu from './WinMenu.js';

class WinFactory extends GUIFactory {
    createButton() {
        return new WinButton();
    }

    createMenu() {
        return new WinMenu();
    }
}

export default WinFactory;

File: hw/cañarte/u3/HW28SingletonAndAbstractFactory/Python/AbstractFactory/controller/WinFactory.py
import sys
import os
# Saray Cañarte, codecreators, DCCO - ESPE

sys.path.append(os.path.abspath(os.path.join(os.path.dirname(__file__), '..')))
from controller.GUIFactory import GUIFactory
from controller.WinButton import WinButton
from controller.WinMenu import WinMenu
class WinFactory(GUIFactory):
    def create_button(self) -> WinButton:
        return WinButton()

    def create_menu(self) -> WinMenu:
        return WinMenu()
